src/transformers/configuration_utils.py

```diff
@@ -1036,6 +1036,7 @@ def _get_non_default_generation_parameters(self) -> Dict[str, Any]:
             if decoder_config is not self:
                 default_config = decoder_config.__class__()
             else:
+                default_config = None
                 decoder_config = None
```

Comment:
I think that there was a typo in this [PR](https://github.com/huggingface/transformers/pull/33219/files) and we should only set `default_config = None` and remove `decoder_config = None` cc @gante